lib/cosmos/gui/text/ruby_editor.rb

```diff
@@ -99,16 +99,16 @@ def self.create_format(color = Cosmos::BLACK, style='')
           ["/.*/", 0, STYLES['string']],
           # Double-quoted string, possibly containing escape sequences
           ['"([^"\\\\]|\\\\.)*"', 0, STYLES['string']],
-          # Match interpolated strings "blah #{code} blah"
-          ['(\#\\{)[^\\}]*\\}', 1, STYLES['brace']],
-          ['\#\\{[^\\}]*(\\})', 1, STYLES['brace']],
-          ['\#\\{([^\\}]*)\\}', 1, STYLES['normal']],
           # Single-quoted string, possibly containing escape sequences
           ["'[^'\\\\]*(\\.[^'\\\\]*)*'", 0, STYLES['string']],
           # Back-tick string, possibly containing escape sequences
           ["`[^`\\\\]*(\\.[^`\\\\]*)*`", 0, STYLES['string']],
+          # Match interpolated strings "blah #{code} blah"
+          ['(\#\\{)[^\\}]*\\}', 1, STYLES['brace']],
+          ['\#\\{[^\\}]*(\\})', 1, STYLES['brace']],
+          ['\#\\{([^\\}]*)\\}', 1, STYLES['normal']],
```

Comment:
Matching the interpolated strings should happen AFTER we identify the strings. Not sure why the order was different.